src/plugins/intel_gpu/src/graph/impls/ocl/non_zero.cpp

```diff
@@ -0,0 +1,117 @@
+        // if (arg.can_be_optimized()) {
+        //     return new count_nonzero_impl(arg, {});
+        // }
+
+        auto nonzero_params = get_default_params<kernel_selector::count_nonzero_params>(arg);
+        auto nonzero_optional_params =
+            get_default_optional_params<kernel_selector::count_nonzero_optional_params>(arg.get_program());
+
+        auto& kernel_selector = kernel_selector::count_nonzero_kernel_selector::Instance();
+        auto best_kernels = kernel_selector.GetBestKernels(nonzero_params, nonzero_optional_params);
+
+        CLDNN_ERROR_BOOL(arg.id(),
+                         "Best_kernel.empty()",
+                         best_kernels.empty(),
+                         "Cannot find a proper kernel with this arguments");
+
+        auto count_nonzero = new count_nonzero_impl(arg, best_kernels[0]);
+
+        return count_nonzero;
+    }
+};
+
+struct gather_nonzero_impl : typed_primitive_impl_ocl<gather_nonzero> {
+    using parent = typed_primitive_impl_ocl<gather_nonzero>;
+    using parent::parent;
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<gather_nonzero_impl>(*this);
+    }
+
+public:
+    static primitive_impl* create(const gather_nonzero_node& arg) {
+        // if (arg.can_be_optimized()) {
+        //     return new gather_nonzero_impl(arg, {});
+        // }
+
+        auto nonzero_params = get_default_params<kernel_selector::gather_nonzero_params>(arg);
+        auto nonzero_optional_params =
+            get_default_optional_params<kernel_selector::gather_nonzero_optional_params>(arg.get_program());
+
+        nonzero_params.inputs.push_back(convert_data_tensor(arg.input(1).get_output_layout()));
+
+        auto& kernel_selector = kernel_selector::gather_nonzero_kernel_selector::Instance();
+        auto best_kernels = kernel_selector.GetBestKernels(nonzero_params, nonzero_optional_params);
+
+        CLDNN_ERROR_BOOL(arg.id(),
+                         "Best_kernel.empty()",
+                         best_kernels.empty(),
+                         "Cannot find a proper kernel with this arguments");
+
+        auto gather_nonzero = new gather_nonzero_impl(arg, best_kernels[0]);
+
+        return gather_nonzero;
+    }
+};
+
+namespace detail {
+
+attach_count_nonzero_impl::attach_count_nonzero_impl() {
+    implementation_map<count_nonzero>::add(impl_types::ocl, count_nonzero_impl::create, {
+        std::make_tuple(data_types::f32, format::bfyx),
+        std::make_tuple(data_types::f16, format::bfyx),
+        std::make_tuple(data_types::i32, format::bfyx),
+        std::make_tuple(data_types::f32, format::bfzyx),
+        std::make_tuple(data_types::f16, format::bfzyx),
+        std::make_tuple(data_types::i32, format::bfzyx),
+        std::make_tuple(data_types::f32, format::bfwzyx),
+        std::make_tuple(data_types::f16, format::bfwzyx),
+        std::make_tuple(data_types::i32, format::bfwzyx),
```

Comment:
is there any reason not to support i8?

Reply:
i8 and u8 are added in the list of supported data types.